### DISCUSSION TITLE
Why mirrord 3.82.0 didn't work on ubuntu 22.04 with minikube

### DISCUSSION THREAD

**Discussion by huangj268** [metalbear-co/mirrord#2171] [Q&A]
Why mirrord didn't create a file in pod? It created a file in the local machine.

minikube version
minikube version: v1.30.1

cat a.py
with open("/tmp/test", "w") as file:
    file.write("hj")

mirrord -V
mirrord 3.82.0

mirrord exec --target pod/sdp-0 python3 a.py
⠐ mirrord exec
    x version check
    ✓ ready to launch process
      ✓ layer extracted
      ✓ operator not found
      ✓ agent pod created
      ✓ pod is ready               

log of agent
agent ready - version 3.82.0
2024-01-12T02:19:46.082762Z  WARN ThreadId(06) mirrord_agent::outgoing: interceptor_task -> no messages left
2024-01-12T02:19:46.082769Z  WARN ThreadId(07) mirrord_agent::outgoing::udp: interceptor_task -> no messages left

**Comment by aviramha** [comment]
Hi,
There are two issues -
1. mirrord by default is read only, meaning any file opened for writing would be local
2. `/tmp` is a path we read and write local by default, since it's not common to have it.

If you'd like the default mode to be rw, you can specify
```json
{
    "feature": {
        "fs":"write"
    }
}
```
If you'd like to write to tmp, you need to have:
```json
{
    "feature": {
        "fs": {
            "read_write": ["/tmp"]
        }
    }
}
```

**Reply by huangj268** [reply]
UDP isn't working. The local process udp_receiver.py can't receive UDP packets.

----------Local
mirrord -V
mirrord 3.84.1
mirrord exec --target pod/sdp-0 python3 udp_receiver.py
⠈ mirrord exec
    x version check
    ✓ ready to launch process
      ✓ layer extracted
      ✓ operator not found
      ✓ agent pod created
      ✓ pod is ready                                                                                            hj

cat udp_receiver.py 
import socket

# Set the UDP broadcast address and port
BROADCAST_ADDR = '10.244.255.255'
BROADCAST_PORT = 5000

# Create a UDP socket
sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)

# Bind the socket to the broadcast address and port
sock.bind((BROADCAST_ADDR, BROADCAST_PORT))

print('hj')
# Receive data packets
while True:
    data, addr = sock.recvfrom(1024)
    print(f'Received data from {addr}: {data.decode()}')

# Close the socket
sock.close()

---------Pod
cat udp_sender.py 
import socket

# Set the UDP broadcast address and port
BROADCAST_ADDR = '10.244.255.255'
BROADCAST_PORT = 5000

# Create a UDP socket
sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)

# Set the message to send
message = b'Hello, World!'

# Send the message to the broadcast address and port
sock.sendto(message, (BROADCAST_ADDR, BROADCAST_PORT))

# Close the socket
sock.close()

**Reply by aviramha** [reply]
Hi, receiving udp packets from remote isn't supported at the moment. If you'd like us to add support please send a feature request.
